.builder/actions/crt-ci-test.py

```diff
@@ -27,29 +27,89 @@ def _write_secret_to_temp_file(self, env, secret_name):
 
         return filename
 
+    def _build_and_run_eventstream_echo_server(self, env):
+        java_sdk_dir = None
+        proc = None
+
+        try:
+            env.shell.exec(["mvn", "--version"], check=True)
+
+            # maven is installed, so this is a configuration we can start an event stream echo server
+            java_sdk_dir = env.shell.mktemp()
+
+            env.shell.exec(["git", "clone", "https://github.com/aws/aws-iot-device-sdk-java-v2"], working_dir=java_sdk_dir, check=True)
+
+            sdk_dir = os.path.join(java_sdk_dir, "aws-iot-device-sdk-java-v2", "sdk")
+            env.shell.pushd(sdk_dir)
+
+            try:
+                env.shell.exec(["mvn", "compile"], check=True)
+                env.shell.exec(["mvn", "test", "-DskipTests=true"], check=True)
+                env.shell.exec(["mvn", "dependency:build-classpath", "-Dmdep.outputFile=classpath.txt"], check=True)
+
+                with open('classpath.txt', 'r') as file:
+                    classpath = file.read()
+
+                test_class_path = os.path.join(sdk_dir, "target", "test-classes")
+                target_class_path = os.path.join(sdk_dir, "target", "classes")
+                directory_separator = os.pathsep
+
+                echo_server_command = ["java", "-classpath", f"{test_class_path}{directory_separator}{target_class_path}{directory_separator}{classpath}", "software.amazon.awssdk.eventstreamrpc.echotest.EchoTestServiceRunner", "127.0.0.1", "8033"]
+
+                print(f'Echo server command: {echo_server_command}')
+
+                # bypass builder's exec wrapper since it doesn't allow for background execution
+                proc = subprocess.Popen(echo_server_command)
```

Comment:
Check out how Dengke set up the background process to get killed when builder exited:
https://github.com/awslabs/aws-c-http/blob/270f0968fd6b5317b8dd714b2bbc6fd04976dbcf/.builder/action/local-server-setup.py#L43